subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -20,12 +20,14 @@ LOG_MODULE_REGISTER(net_sock_addr, CONFIG_NET_SOCKETS_LOG_LEVEL);
 #include <net/socket_offload.h>
 #include <syscall_handler.h>
 
-#define AI_ARR_MAX	2
```

Comment:
IMHO it would be better to have
```
#define AI_ARR_MAX CONFIG_DNS_RESOLVER_AI_MAX_ENTRIES
```
here instead. Then we do not need to change the relevant and existing code below.